In [4]:
import numpy as np

In [5]:
from sklearn.metrics import mean_squared_error

np.sqrt(mean_squared_error([1000], [1400]))

400.0

In [6]:
np.sqrt(mean_squared_error([1000], [600]))

400.0

In [7]:
from sklearn.metrics import mean_squared_log_error

np.sqrt(mean_squared_log_error([1000], [1400]))

0.3361867670217862

In [8]:
from sklearn.metrics import mean_squared_log_error

np.sqrt(mean_squared_log_error([1000], [600]))

0.5101598447800129

使用 RMSLE 的好处一：

+ 假如真实值为 $1000$，若果预测值是 $600$，那么 $RMSE=400$， $RMSLE=0.510$；

+ 假如真实值为 $1000$，若预测结果为 $1400$， 那么 $RMSE=400$， $RMSLE=0.336$。

可以看出来在均方根误差相同的情况下，预测值比真实值小这种情况的错误比较大，即对于预测值小这种情况惩罚较大。

使用 RMSLE 的好处二：

直观的经验是这样的，当数据当中有少量的值和真实值差值较大的时候，使用log函数能够减少这些值对于整体误差的影响。



